table_engine/src/remote/model.rs

```diff
@@ -148,22 +149,33 @@ impl TryFrom<ceresdbproto::remote_engine::WriteRequest> for WriteRequest {
     ) -> std::result::Result<Self, Self::Error> {
         let table_identifier = pb.table.context(EmptyTableIdentifier)?;
         let row_group_pb = pb.row_group.context(EmptyRowGroup)?;
-        let table_schema: Schema = row_group_pb
-            .table_schema
-            .context(EmptyTableSchema)?
-            .try_into()
-            .box_err()
-            .context(ConvertTableSchema)?;
-        let row_group = if row_group_pb.version == ENCODE_ROWS_WITH_AVRO {
-            avro::avro_rows_to_row_group(table_schema, &row_group_pb.rows)
-                .box_err()
-                .context(ConvertRowGroup)?
-        } else {
-            UnsupportedConvertRowGroup {
-                version: row_group_pb.version,
+        let rows = row_group_pb.rows.context(EmptyRowGroup)?;
+        let row_group = match rows {
+            Arrow(v) => {
+                ensure!(
+                    !v.record_batches.is_empty(),
+                    InvalidRecordBatchNumber {
+                        batch_num: v.record_batches.len(),
+                    }
+                );
```

Comment:
- Why can't the `record_batches` be empty?
- The error message for the `InvalidRecordBatchNumber` says `Invalid record batches number in the response, expect only one`, which seems wrong with the exact meaning here.